### DISCUSSION TITLE
Improve types for `Headers.get`

### DISCUSSION THREAD

**Discussion by edmorley** [encode/starlette#1795] [Potential Issue] [answered]
Currently `Headers.get()` is defined as:
```python
def get(self, key: str, default: typing.Any = None) -> typing.Any:
```

https://github.com/encode/starlette/blob/d81d8c8d3cdd9fd782bcdc0bf1442409e2d8d902/starlette/datastructures.py#L543

The use of `typing.Any` means the type checker won't validate the use of the return value.

It seems that something like this work give better results?

```python
T = typing.TypeVar("T")

def get(self, key: str, default: T = None) -> typing.Union[str, T]:
```

**Comment by Kludex** [comment]
This https://github.com/encode/starlette/pull/1650 ?

**Reply by edmorley** [reply]
That seems related, however I don't see that PR modifying `Headers`, which defines `get()` itself (it doesn't inherit `get()` from `MultiDict`).

**Comment by Kludex** [comment, marked as answer]
- PR to solve this was created on #1810
